crates/libs/bindgen/src/classes.rs

```diff
@@ -51,14 +52,20 @@ fn gen_class(def: &TypeDef, gen: &Gen) -> TokenStream {
                     quote! {}
                 };
 
+                let factory_cache_method = if let Some(library) = gen.class_map.get(&class_name.as_str()) {
+                    format_token!(r#"new_with_library("{library}")"#)
+                } else {
+                    format_token!("new()")
+                };
+
```

Comment:
Maybe hoist this out of this scope so we can share the code gen and avoid the redundancy below.